python/src/trezorlib/cli/ethereum.py

```diff
@@ -164,34 +167,168 @@ def _format_access_list(
     ]
 
 
+def _get_ethereum_definitions(
+    definitions_zip: Optional[pathlib.Path] = None,
+    network_def_file: Optional[BinaryIO] = None,
+    token_def_file: Optional[BinaryIO] = None,
+    download_definitions: bool = False,
+    chain_id: Optional[int] = None,
+    slip44_hardened: Optional[int] = None,
+    token_address: Optional[str] = None,
+) -> ethereum.messages.EthereumDefinitions:
+    count_of_options_used = sum(
+        bool(o)
+        for o in (
+            definitions_zip,
+            (network_def_file or token_def_file),
+            download_definitions,
+        )
+    )
+    if count_of_options_used > 1:
+        raise click.ClickException(
+            "More than one mutually exclusive option for definitions was used. See --help for more info."
+        )
+
+    slip44 = None
+    if slip44_hardened is not None:
+        slip44 = UH_(slip44_hardened)
+
+    defs = ethereum.messages.EthereumDefinitions()
+    if definitions_zip is not None:
+        if chain_id is not None or slip44 is not None:
+            defs.encoded_network = ethereum.get_definition_from_zip(
+                definitions_zip,
+                ethereum.get_network_definition_path(chain_id, slip44),
+            )
+        if chain_id is not None and token_address is not None:
+            defs.encoded_token = ethereum.get_definition_from_zip(
+                definitions_zip,
+                ethereum.get_token_definition_path(chain_id, token_address),
+            )
+    elif network_def_file is not None or token_def_file is not None:
+        if network_def_file is not None:
+            with network_def_file:
+                defs.encoded_network = network_def_file.read()
+        if token_def_file is not None:
+            with token_def_file:
+                defs.encoded_token = token_def_file.read()
+    elif download_definitions:
+        if chain_id is not None or slip44 is not None:
+            defs.encoded_network = ethereum.download_from_url(
+                ethereum.get_network_definition_url(chain_id, slip44)
+            )
+        if chain_id is not None and token_address is not None:
+            defs.encoded_network = ethereum.download_from_url(
```

Comment:
Shouldn't this be `encoded_token`? Could be copy/paste mistake

Reply:
affected code was removed